tests/generic_config_updater/test_syslog.py

```diff
@@ -0,0 +1,267 @@
+
+    expected_content_list = ["[{}]".format(dummy_syslog_server_v4), "[{}]".format(dummy_syslog_server_v6)]
+    expect_res_success_syslog(duthost, expected_content_list, [])
+
+    delete_tmpfile(duthost, tmpfile)
+
+@pytest.mark.parametrize("op, dummy_syslog_server_v4, dummy_syslog_server_v6", [
+    ("add", "10.0.0.587", "cc98:2008::1"),
+    ("add", "10.0.0.5", "cc98:2008::xyz"),
+    ("remove", "10.0.0.6", "cc98:2008:1"),
+    ("remove", "10.0.0.5", "cc98:2008::2")
+])
+def test_syslog_server_tc3_xfail(duthost, setup_env, op,
+        dummy_syslog_server_v4, dummy_syslog_server_v6):
+    """ Test expect fail testcase
+
+    ("add", "10.0.0.587", "cc98:2008::1"), ADD Invalid IPv4 address
+    ("add", "10.0.0.5", "cc98:2008::xyz"), ADD Invalid IPv6 address
+    ("remove", "10.0.0.6", "cc98:2008:1"), REMOVE Unexist IPv4 address
+    ("remove", "10.0.0.5", "cc98:2008::2") REMOVE Unexist IPv6 address
+    """
+    json_patch = [
+        {
+            "op": "{}".format(op),
+            "path": "/SYSLOG_SERVER/{}".format(dummy_syslog_server_v4),
+            "value": {}
+        },
+        {
+            "op": "{}".format(op),
+            "path": "/SYSLOG_SERVER/{}".format(dummy_syslog_server_v6),
+            "value": {}
+        }
+    ]
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+    expect_op_failure(output)
+
+    delete_tmpfile(duthost, tmpfile)
+
+@pytest.mark.parametrize("op, dummy_syslog_server_v4, dummy_syslog_server_v6", [
+    ("remove", "10.0.0.5", "cc98:2008::1")
+])
+def test_syslog_server_tc4_remove(duthost, setup_env, op,
+        dummy_syslog_server_v4, dummy_syslog_server_v6):
+    """ Remove v4 and v6 syslog server
+
+    admin@vlab-01:~$ show runningconfiguration syslog
+    Sample output:
+    Syslog Servers
+    ----------------
+    """
+    json_patch = [
+        {
+            "op": "{}".format(op),
+            "path": "/SYSLOG_SERVER"
+        }
+    ]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+    expect_op_success_and_reset_check(duthost, output, 'rsyslog-config', SYSLOG_THRESHOLD, SYSLOG_INTERVAL, 0)
+
+    unexpected_content_list = ["[{}]".format(dummy_syslog_server_v4), "[{}]".format(dummy_syslog_server_v6)]
+    expect_res_success_syslog(duthost, [], unexpected_content_list)
+
+    delete_tmpfile(duthost, tmpfile)
+
+def test_syslog_server_tc5_add_to_max(duthost, setup_env):
+    """ Test syslog server max
+
+    admin@vlab-01:~$ show runningconfiguration syslog
+    Sample output:
+    Syslog Servers
+    ----------------
+    [10.0.0.1]
+    ...
+    [10.0.0.SYSLOG_MAX_SERVER]
+    """
+    if SYSLOG_MAX_SERVER == -1:
+        pytest.skip("SYSLOG_MAX_SERVER is not set")
+
+    syslog_servers = ["10.0.0.{}".format(i) for i in range(1, SYSLOG_MAX_SERVER+1)]
```

Comment:
>SYSLOG_MAX_SERVER [](http://example.com/codeflow?start=62&length=17)

What if SYSLOG_MAX_SERVER >= 256? invalid IP?

Reply:
Where does 256 come from? It seems we dont have a max from sonic-syslog.yang.
Invalid IP was supported in tc4.

Reply:
Got your point. I set a test max upper limit. 
I will change related code if the syslog server maximum in yang is over that limit.